quick-protobuf/tests/rust_protobuf/v2/test_default_values_test.rs

```diff
@@ -1,44 +1,122 @@
+use std::borrow::Cow;
 use std::f32;
 use std::f64;
 
 use super::test_default_values::*;
+use super::test_default_values_dont_use_cow::*;
 use quick_protobuf::*;
 
 #[test]
 fn test_default_value_simple() {
     let bytes = &[];
     let mut reader = BytesReader::from_bytes(bytes);
-    let d = TestDefaultValues::from_reader(&mut reader, bytes).unwrap();
-    assert_eq!(1.0, d.double_field);
-    assert_eq!(2.0, d.float_field);
-    assert_eq!(3, d.int32_field);
-    assert_eq!(4, d.int64_field);
-    assert_eq!(5, d.uint32_field);
-    assert_eq!(6, d.uint64_field);
-    assert_eq!(7, d.sint32_field);
-    assert_eq!(8, d.sint64_field);
-    assert_eq!(9, d.fixed32_field);
-    assert_eq!(10, d.fixed64_field);
-    assert_eq!(11, d.sfixed32_field);
-    assert_eq!(12, d.sfixed64_field);
-    assert_eq!(true, d.bool_field);
-    assert_eq!("abc\n22", d.string_field);
-    assert_eq!(b"cde\n33".to_vec(), d.bytes_field.to_vec());
-    assert!(EnumForDefaultValue::TWO.eq(&d.enum_field));
+    let d = TestDefaultValuesOptional::from_reader(&mut reader, bytes).unwrap();
+    assert_eq!(1.0, d.double_field.unwrap_or(TestDefaultValuesOptional::DEFAULT_double_field));
```

Comment:
Shouldn't this assert be that it is `None` and a separate assert that `DEFAULT_double_field == 1.0`?

Reply:
Yep, I think I originally wanted to test exactly what the user would write (i.e. `obj.field.unwrap_or(Class::DEFAULT_field)` so I just wrote it in verbatim

Have changed it though since your version makes more sense haha